src/rules/keyframes.rs

```diff
@@ -33,6 +35,67 @@ pub struct KeyframesRule<'i> {
   pub loc: Location,
 }
 
+/// KeyframesName
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum KeyframesName<'i> {
+  /// `<custom-ident>` of a `@keyframes` name.
+  #[cfg_attr(feature = "serde", serde(borrow))]
+  Ident(CustomIdent<'i>),
+
+  /// `<string>` of a `@keyframes` name.
+  #[cfg_attr(feature = "serde", serde(borrow))]
+  Custom(CowArcStr<'i>),
+}
+
+impl<'i> Parse<'i> for KeyframesName<'i> {
+  fn parse<'t>(input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i, ParserError<'i>>> {
+    match input.next()?.clone() {
+      Token::Ident(ref s) => {
+        // CSS-wide keywords without quotes throws an error.
+        match_ignore_ascii_case! { &*s,
+          "none" | "initial" | "inherit" | "unset" | "default" | "revert" | "revert-layer" => {
+            Err(input.new_unexpected_token_error(Token::Ident(s.clone())))
+          },
+          _ => {
+            Ok(KeyframesName::Ident(CustomIdent(s.into())))
+          }
+        }
+      }
+
+      Token::QuotedString(ref s) => Ok(KeyframesName::Custom(s.into())),
+      t => return Err(input.new_unexpected_token_error(t.clone())),
+    }
+  }
+}
+
+impl<'i> ToCss for KeyframesName<'i> {
+  fn to_css<W>(&self, dest: &mut Printer<W>) -> Result<(), PrinterError>
+  where
+    W: std::fmt::Write,
+  {
+    match self {
+      KeyframesName::Ident(ident) => {
+        dest.write_ident(ident.0.as_ref())?;
+      }
+      KeyframesName::Custom(s) => {
+        // CSS-wide keywords and `none` cannot remove quotes.
+        match_ignore_ascii_case! { &*s,
+          "none" | "initial" | "inherit" | "unset" | "default" | "revert" | "revert-layer" => {
+            dest.write_char('"')?;
+            dest.write_str(s.as_ref())?;
+            dest.write_char('"')?;
```

Comment:
You should use the `serialize_string` function from the `cssparser` crate for this, as it will handle escaping quotes and such.

Reply:
Done